pkg/diagnose/scan_linux.go

```diff
@@ -0,0 +1,60 @@
+//go:build linux
+// +build linux
+
+package diagnose
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/dustin/go-humanize"
+	"github.com/euank/go-kmsg-parser/v3/kmsgparser"
+
+	nvidia_sxid "github.com/leptonai/gpud/components/accelerator/nvidia/error/sxid"
+	nvidia_xid "github.com/leptonai/gpud/components/accelerator/nvidia/error/xid"
+	"github.com/leptonai/gpud/pkg/log"
+)
+
+func scanKmsg(ctx context.Context) {
+	fmt.Printf("%s scanning kmsg\n", inProgress)
+	kl, err := kmsgparser.NewParser(kmsgparser.WithNoFollow())
+	if err != nil {
+		log.Logger.Warnw("error creating kmsg parser", "error", err)
+		return
+	}
+	defer kl.Close()
+
+	ch := make(chan kmsgparser.Message, 1024)
+	if err := kl.Parse(ch); err != nil {
```

Comment:
Parse is blocked, so we should use a separate goroutine for it.  
We can try use [errgroup](golang.org/x/sync/errgroup) to make the code clean..

```go
ch := make(chan kmsgparser.Message, 1024)
group, _ := errgroup.WithContext(c)
group.Go(func() error{
   return kl.Parse(ch)
})

cnt := 0
for msg := range ch {
    cnt++
	ts := humanize.RelTime(msg.Timestamp, now, "ago", "from now")

	if found := nvidia_xid.Match(msg.Message); found != nil {
		fmt.Printf("[XID found] (%s) %q\n", ts, msg.Message)
	}
	if found := nvidia_sxid.Match(msg.Message); found != nil {
		fmt.Printf("[SXID found] (%s) %q\n", ts, msg.Message)
	}
}

err := group.Wait()
if err != nil {
   log.Logger.Errorw("failed to parse kmsg", "error", err) 
}

```